example/src/list.cpp

```diff
@@ -677,4 +677,187 @@ IS_SAME(
+  metal::detail::_min_element<metal::lambda<std::is_void>>::template combiner,
+  void, some_extra_argument, char
+>;
+)
+
+HIDE(
+template<class, class>
+using good_comp = metal::false_;
+
+static_assert(
+  metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<good_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class...>
+using good_comp = metal::false_;
+
+static_assert(
+  metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<good_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class, class...>
+using good_comp = metal::false_;
+
+static_assert(
+  metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<good_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class, class>
+struct good_lazy_comp {
+  using type = metal::false_;
+};
+
+template<class... args>
+using good_comp = typename good_lazy_comp<args...>::type;
+
+static_assert(
+  metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<good_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class, class, class...>
+struct good_lazy_comp {
+  template<class...>
+  using type = metal::false_;
+};
+
+template<class... args>
+using good_comp = typename good_lazy_comp<args...>::template type<args...>;
+
+static_assert(
+  metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<good_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class>
+struct bad_lazy_comp {
+  using type = metal::false_;
+};
+
+template<class... args>
+using bad_comp = typename bad_lazy_comp<args...>::type;
+
+static_assert(
+  !metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<bad_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class>
+using bad_comp = metal::false_;
+
+static_assert(
+  !metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<bad_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class, class, class>
+using bad_comp = metal::false_;
+
+static_assert(
+  !metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<bad_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class, class, class, class...>
+using bad_comp = metal::false_;
+
+static_assert(
+  !metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<bad_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class, class>
+struct bad_comp;
+
+static_assert(
+  !metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<bad_comp>
+  >{}, "");
+)
+
+HIDE(
+template<class, class>
+struct bad_comp {};
+
+static_assert(
+  !metal::is_invocable<
+  metal::lambda<metal::min_element>,
+  metal::numbers<4, 42, 17, -7, 0>,
+  metal::lambda<bad_comp>
+  >{}, "");
+)
```

Comment:
You probably realize this, but this file is intended to host examples that go in the doc, unit tests should be defined in therms of the preprocessor based suite.

Reply:
Yes, you are right.

This is just more convenient to me, and I also wanted to show it to you. It is definitely not intented to be merged.